### PR TITLE
feat: Add CSV Destination plugin

### PR DESCRIPTION
**Pull request by yevgenypats**
I rebased incorrectly previous PR.

### CODE REVIEW

File: plugins/destination/csv/client/migrate.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	for _, t := range tables {
+		filePath := path.Join(c.csvSpec.Directory, t.Name+".csv")
+		//nolint:gocritic,revive
+		if _, err := os.Stat(filePath); err == nil {
+			f, err := os.Open(filePath)
+			if err != nil {
+				return err
+			}
+			//nolint:revive
+			defer f.Close()
```
Comment (disq) [inline plugins/destination/csv/client/migrate.go:36]: I would wrap the whole block inside a fn (anonymous or a separate function) and remove the `nolint`. No reason to keep files open more than necessary.

**Review by disq — CHANGES_REQUESTED** [review]

File: plugins/destination/csv/client/client.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+type tableWriter struct {
+	writer *csv.Writer
+	file   *os.File
+	count  uint64
+}
+
+func New(ctx context.Context, logger zerolog.Logger, spec specs.Destination) (plugins.DestinationClient, error) {
+	if spec.WriteMode != specs.WriteModeAppend {
+		return nil, fmt.Errorf("csv destination only supports append mode only")
```
Comment (hermanschaaf) [inline plugins/destination/csv/client/client.go:67]:
```suggestion
		return nil, fmt.Errorf("csv destination only supports append mode")
```

File: plugins/destination/csv/client/client_test.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	}
+	if client == nil {
+		t.Fatal("client is nil")
+	}
+	if err := client.Close(ctx); err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = New(ctx, getTestLogger(t), specs.Destination{})
+	if err.Error() != "csv destination only supports append mode only" {
```
Comment (hermanschaaf) [inline plugins/destination/csv/client/client_test.go:39]:
```suggestion
	if err.Error() != "csv destination only supports append mode" {
```

File: plugins/destination/csv/main.go [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import (
+	"github.com/cloudquery/cloudquery/plugins/destination/csv/client"
+	"github.com/cloudquery/cloudquery/plugins/destination/csv/resources/plugin"
+	"github.com/cloudquery/plugin-sdk/plugins"
+	"github.com/cloudquery/plugin-sdk/serve"
+)
+
+const (
+	// TODO: add sentry DSN
+	sentryDSN = ""
```
Comment (hermanschaaf) [inline plugins/destination/csv/main.go]: Let's remember to do this before merging ☝️

**Review by hermanschaaf — CHANGES_REQUESTED** [review]

File: plugins/destination/csv/main.go (hunk 4 above)
Comment (yevgenypats) [reply plugins/destination/csv/main.go]: Done

File: plugins/destination/csv/client/migrate.go (hunk 1 above)
Comment (yevgenypats) [reply plugins/destination/csv/client/migrate.go]: Good idea.

**Review by hermanschaaf — APPROVED** [review]

**Pull request merged**